pinot-core/src/main/java/org/apache/pinot/core/segment/processing/utils/SegmentProcessorUtils.java

```diff
@@ -66,42 +66,58 @@ public static Schema convertPinotSchemaToAvroSchema(org.apache.pinot.spi.data.Sc
       if (fieldSpec.isSingleValueField()) {
         switch (dataType) {
           case INT:
-            fieldAssembler = fieldAssembler.name(name).type().intType().noDefault();
+            fieldAssembler =
+                fieldAssembler.name(name).type().unionOf().intType().and().nullType().endUnion().noDefault();
```

Comment:
What's the purpose of making it union? Also, please add that into the PR description.

Reply:
By default, Avro doesn't allow you to have `null` value unless you use the type of `union(int, null)`. For instance, let's assume that we added a column to the schema while the segment that we are reading is generated before and doesn't contain the new column. In this case, if we read the row, the new column value will always be `null`. Without using a union type, our segment processor will throw the exception and cannot generate the segment.  I will add the comments.